app/assets/javascripts/hubstats/application.js

```diff
@@ -14,45 +14,83 @@
 //= require jquery_ujs
 //= require select2
 //= require hubstats/bootstrap
+//= require bootstrap-datepicker
 //= require_tree .
 
 $(document).ready( function() {
+  defaultDateRange()
   setDateRange()
 });
 
+function defaultDateRange() {
+  $('.input-daterange').datepicker({
+    "todayHighlight": true,
+    "endDate": "Today",
+    "todayBtn": true
+  });
+};
+
 function setDateRange() {
-  var index = readCookie("hubstats_index") || 2;
-  var timer = document.getElementById("time-select");
+  if ((readCookie("hubstats_index") === "null~~null") || (readCookie("hubstats_index") === null)) {
+    var index = getDefaultDateRange();
+  } else {
+    var index = readCookie("hubstats_index")
+  }
 
-  timer.selectedIndex = index;
+  var timer = document.getElementById("submitDateRange");
+  dates = index.split("~~")
+  $('.input-daterange').find('[name="start"]').datepicker('update', new Date(dates[0]));
+  $('.input-daterange').find('[name="end"]').datepicker('update', new Date(dates[1]));
 
-  timer.onchange = function() {
-    createCookie("hubstats_index",this.selectedIndex,1);
+  timer.onclick = function() {
+    start_date = $('.input-daterange').find('[name="start"]').datepicker('getDate');
+    end_date = $('.input-daterange').find('[name="end"]').datepicker('getDate');
+    createCookie("hubstats_index", start_date, end_date, 1);
     window.location.reload();
   };
 };
 
-function createCookie(name,value,days) {
-    if (days) {
-        var date = new Date();
-        date.setTime(date.getTime()+(days*24*60*60*1000));
-        var expires = "; expires="+date.toGMTString();
-    }
-    else var expires = "";
-    document.cookie = name+"="+value+expires+"; path=/";
+function createCookie(name,value1,value2,days) {
```

Comment:
Is this function used any where else in the code?

A cookie is just a single string value and the separate is really specific to the date picker.  I don't think we should put the separator logic in this function incase we need to introduce other cookies for the project at some point.  Create and read cookie should stay simple and do their one job of reading/writing a string value.  Move the datepicker separator logic to the datepicker code.